### ISSUE TITLE
[package] arrow/15.0.0: user options are not respected

### ISSUE THREAD

**Issue by wgtmac** [conan-io/conan-center-index#22506]
### Description

A set of self.options.xxx options are initialized in config_options: https://github.com/conan-io/conan-center-index/blob/master/recipes/arrow/all/conanfile.py#L161-L175. However, this is a misuse. The config_options() method is used to configure or constrain the available options in a package before assigning them a value. The config_options() method executes:
- Before calling the configure() method.
- Before assigning the options values.
- After settings are already defined.See 

Check https://docs.conan.io/2.0/reference/conanfile/methods/config_options.html for detail.

I've opened a PR in the arrow community: https://github.com/apache/arrow/pull/39729. It would be good to make it consistent to the upstream.

### Package and Environment Details

* Package Name/Version: **arrow/15.0.0**
* Operating System+version: **MacOS 12.5.1**
* Compiler+version: **omebrew clang version 17.0.4**
* Conan version: **conan 2.0.14**
* Python version: **Python 3.11.6**

### Conan profile

[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=17
os=Macos

### Steps to reproduce

conan create . --profile:host default --profile:build default --build=missing --version 15.0.0 -o parquet=True -o with_brotli=True -o with_bz2=True -o with_flight_rpc=True -o with_glog=True -o with_jemalloc=True -o with_json=True -o with_lz4=True -o with_snappy=True -o with_zstd=True

### Logs

<details><summary>Click to expand log</summary>

```
======== Exporting recipe to the cache ========
arrow/15.0.0: Exporting package recipe: /Users/gangwu/Projects/conan-center-index/recipes/arrow/all/conanfile.py
arrow/15.0.0: exports: File 'conandata.yml' found. Exporting it...
arrow/15.0.0: Calling export_sources()
arrow/15.0.0: Copied 1 '.py' file: conanfile.py
arrow/15.0.0: Copied 1 '.yml' file: conandata.yml
arrow/15.0.0: Copied 1 '.cmake' file: conan_cmake_project_include.cmake
arrow/15.0.0: Exported to cache folder: /Users/gangwu/.conan2/p/arrowe946f56c35eba/e
arrow/15.0.0: Exported: arrow/15.0.0#1528e9021cdbb2e34c1ca8bdd948fe5a (2024-01-23 08:17:55 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=17
os=Macos
[options]
parquet=True
with_brotli=True
with_bz2=True
with_flight_rpc=True
with_glog=True
with_jemalloc=True
with_json=True
with_lz4=True
with_snappy=True
with_zstd=True

Profile build:
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=17
os=Macos


======== Computing dependency graph ========
Graph root
    cli
Requirements
    arrow/15.0.0#1528e9021cdbb2e34c1ca8bdd948fe5a - Cache
    brotli/1.1.0#d56d7bb9ca722942aba17369cb5c0519 - Cache
    bzip2/1.0.8#411fc05e80d47a89045edc1ee6f23c1d - Cache
    gflags/2.2.2#48d1262ffac8d30c3224befb8275a533 - Cache
    glog/0.6.0#a62086b490f39299eed174919966122e - Cache
    jemalloc/5.3.0#1deb9d1c836d40b85554ab3a3f5b352e - Cache
    lz4/1.9.4#1217a03c990b79aa34ed0faede18f534 - Cache
    rapidjson/1.1.0#9f5e39d1f4a6b67d48c6a343c4592f32 - Cache
    snappy/1.1.9#5bcaf46bc878b6403844d1be7b674aa2 - Cache
    xsimd/9.0.1#f93b209d2cb92ac4b73e994e65c66aa1 - Cache
    zstd/1.5.5#93372fe14bb7883bd4de82914e0a1841 - Cache
Build requirements
    autoconf/2.71#53be95d228b2dcb30dc199cb84262d8f - Cache
    automake/1.16.5#058bda3e21c36c9aa8425daf3c1faf50 - Cache
    cmake/3.27.7#6e11610a238b9292dffda14c34bb53e6 - Cache
    m4/1.4.19#b38ced39a01e31fef5435bc634461fd2 - Cache
Resolved version ranges
    cmake/[>=3.16 <4]: cmake/3.27.7

======== Computing necessary packages ========
arrow/15.0.0: Checking 5 compatible configurations
arrow/15.0.0: Compatible configurations not found in cache, checking servers
arrow/15.0.0: 'd8e3991d4b80194e9f67cff30ea18e30b6a29811': compiler.cppstd=17
arrow/15.0.0: 'bbec488f5c8627afa20dd5a7f75f0b6db7fc5d76': compiler.cppstd=20
arrow/15.0.0: 'b090841e23fea5c01e12ba339b56c96678411f9c': compiler.cppstd=gnu20
arrow/15.0.0: '56b6ff26f6b910e21b4cd3a119907f882cca589c': compiler.cppstd=23
arrow/15.0.0: '27c1391d1f2262a8267627ec718060038e3292b5': compiler.cppstd=gnu23
Requirements
    arrow/15.0.0#1528e9021cdbb2e34c1ca8bdd948fe5a:92b89103d00005e6970e71f2ae5d7dfc138f73aa - Build
    brotli/1.1.0#d56d7bb9ca722942aba17369cb5c0519:57511b8eef0b94f2b6b86d5d575920eec716d1a2#775fb46ff282ae0d609266aa4311a064 - Cache
    bzip2/1.0.8#411fc05e80d47a89045edc1ee6f23c1d:0f7d9d719d0146d61ac4f9df12f6b81b35a1754d#b3022265e2d8df8115ebed14e4ecc322 - Cache
    gflags/2.2.2#48d1262ffac8d30c3224befb8275a533:c5bdd2c9bbcd2aa4ca5a5b405d4e650c99af5bd6#eee7f66ca9c5b6cd00c40e178ad17397 - Cache
    glog/0.6.0#a62086b490f39299eed174919966122e:be1d855159be3e48996ef2d34a4eae5ed534d56a#7107c6de85a3af696a6cce6ba23b8c81 - Cache
    jemalloc/5.3.0#1deb9d1c836d40b85554ab3a3f5b352e:a8c3c4f1cc7c037c945b8619d64dd337d3103d10#bd9141c9dd21c95062ade792c9e6977f - Cache
    lz4/1.9.4#1217a03c990b79aa34ed0faede18f534:08e70e1e2e2d840af7b4c965f85dda12d583e803#7c8f790fc58b648511eb2e1140288e02 - Cache
    rapidjson/1.1.0#9f5e39d1f4a6b67d48c6a343c4592f32:da39a3ee5e6b4b0d3255bfef95601890afd80709#c8d3999780598e0fbaed67c22f961d87 - Cache
    snappy/1.1.9#5bcaf46bc878b6403844d1be7b674aa2:dcae45575faf7c83fa9471408fe53403183f02c9#57011e3b26d543f18cc30eb01a457e30 - Cache
    xsimd/9.0.1#f93b209d2cb92ac4b73e994e65c66aa1:da39a3ee5e6b4b0d3255bfef95601890afd80709#7582132cb8d532d695e8c242308191f2 - Cache
    zstd/1.5.5#93372fe14bb7883bd4de82914e0a1841:9a820bda9b10b463da5354540f05bed5747f8cac#9e76c6506e49ea959baf0a210ab543d8 - Cache
Build requirements
    cmake/3.27.7#6e11610a238b9292dffda14c34bb53e6:9e5323c65b94ae38c3c733fe12637776db0119a5#b33ffc06aaa1ca3eb26f45ba1a3e9c54 - Cache
Skipped binaries
    autoconf/2.71, automake/1.16.5, m4/1.4.19

======== Installing packages ========
brotli/1.1.0: Already installed! (1 of 12)
bzip2/1.0.8: Already installed! (2 of 12)
cmake/3.27.7: Already installed! (3 of 12)
cmake/3.27.7: Appending PATH environment variable: /Users/gangwu/.conan2/p/cmakefc220a4e56e94/p/CMake.app/Contents/bin
gflags/2.2.2: Already installed! (4 of 12)
lz4/1.9.4: Already installed! (5 of 12)
rapidjson/1.1.0: Already installed! (6 of 12)
snappy/1.1.9: Already installed! (7 of 12)
xsimd/9.0.1: Already installed! (8 of 12)
zstd/1.5.5: Already installed! (9 of 12)
jemalloc/5.3.0: Already installed! (10 of 12)
glog/0.6.0: Already installed! (11 of 12)
arrow/15.0.0: Calling source() in /Users/gangwu/.conan2/p/arrowe946f56c35eba/s/src
arrow/15.0.0: Downloading 21.5MB apache-arrow-15.0.0.tar.gz

-------- Installing package arrow/15.0.0 (12 of 12) --------
arrow/15.0.0: Building from source
arrow/15.0.0: Package arrow/15.0.0:92b89103d00005e6970e71f2ae5d7dfc138f73aa
arrow/15.0.0: Copying sources to build folder
arrow/15.0.0: Building your package in /Users/gangwu/.conan2/p/b/arrow330afd2a1039a/b
arrow/15.0.0: Calling generate()
arrow/15.0.0: Generators folder: /Users/gangwu/.conan2/p/b/arrow330afd2a1039a/b/build/Release/generators
arrow/15.0.0: CMakeToolchain generated: conan_toolchain.cmake
arrow/15.0.0: CMakeToolchain generated: CMakePresets.json
arrow/15.0.0: Generating aggregated env files
arrow/15.0.0: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
arrow/15.0.0: Calling build()
arrow/15.0.0: Running CMake.configure()
arrow/15.0.0: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/Users/gangwu/.conan2/p/b/arrow330afd2a1039a/b/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/gangwu/.conan2/p/b/arrow330afd2a1039a/p" -DCMAKE_PROJECT_arrow_INCLUDE="/Users/gangwu/.conan2/p/b/arrow330afd2a1039a/b/src/conan_cmake_project_include.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/gangwu/.conan2/p/b/arrow330afd2a1039a/b/src/cpp"
-- Building using CMake version: 3.27.7
-- Using Conan toolchain: /Users/gangwu/.conan2/p/b/arrow330afd2a1039a/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: C++ Standard 17 with extensions ON
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The C compiler identification is AppleClang 13.1.6.13160021
-- The CXX compiler identification is AppleClang 13.1.6.13160021
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Arrow version: 15.0.0 (full: '15.0.0')
-- Arrow SO version: 1500 (full: 1500.0.0)
-- clang-tidy 14 not found
-- clang-format 14 not found
-- Could NOT find ClangTools (missing: CLANG_FORMAT_BIN CLANG_TIDY_BIN)
-- infer not found
-- Found Python3: /opt/homebrew/bin/python3.11 (found version "3.11.6") found components: Interpreter
fatal: not a git repository (or any of the parent directories): .git
-- Found cpplint executable at /Users/gangwu/.conan2/p/b/arrow330afd2a1039a/b/src/cpp/build-support/cpplint.py
-- System processor: arm64
-- Performing Test CXX_SUPPORTS_SVE
-- Performing Test CXX_SUPPORTS_SVE - Success
-- Arrow build warning level: PRODUCTION
-- Build Type: RELEASE
-- Performing Test CXX_LINKER_SUPPORTS_VERSION_SCRIPT
-- Performing Test CXX_LINKER_SUPPORTS_VERSION_SCRIPT - Failed
-- Using SYSTEM approach to find dependencies
-- ARROW_ABSL_BUILD_VERSION: 20211102.0
-- ARROW_ABSL_BUILD_SHA256_CHECKSUM: dcf71b9cba8dc0ca9940c4b316a0c796be8fab42b070bb6b7cab62b48f0e66c4
-- ARROW_AWS_C_AUTH_BUILD_VERSION: v0.6.22
-- ARROW_AWS_C_AUTH_BUILD_SHA256_CHECKSUM: 691a6b4418afcd3dc141351b6ad33fccd8e3ff84df0e9e045b42295d284ee14c
-- ARROW_AWS_C_CAL_BUILD_VERSION: v0.5.20
-- ARROW_AWS_C_CAL_BUILD_SHA256_CHECKSUM: acc352359bd06f8597415c366cf4ec4f00d0b0da92d637039a73323dd55b6cd0
-- ARROW_AWS_C_COMMON_BUILD_VERSION: v0.8.9
-- ARROW_AWS_C_COMMON_BUILD_SHA256_CHECKSUM: 2f3fbaf7c38eae5a00e2a816d09b81177f93529ae8ba1b82dc8f31407565327a
-- ARROW_AWS_C_COMPRESSION_BUILD_VERSION: v0.2.16
-- ARROW_AWS_C_COMPRESSION_BUILD_SHA256_CHECKSUM: 044b1dbbca431a07bde8255ef9ec443c300fc60d4c9408d4b862f65e496687f4
-- ARROW_AWS_C_EVENT_STREAM_BUILD_VERSION: v0.2.18
-- ARROW_AWS_C_EVENT_STREAM_BUILD_SHA256_CHECKSUM: 310ca617f713bf664e4c7485a3d42c1fb57813abd0107e49790d107def7cde4f
-- ARROW_AWS_C_HTTP_BUILD_VERSION: v0.7.3
-- ARROW_AWS_C_HTTP_BUILD_SHA256_CHECKSUM: 07e16c6bf5eba6f0dea96b6f55eae312a7c95b736f4d2e4a210000f45d8265ae
-- ARROW_AWS_C_IO_BUILD_VERSION: v0.13.14
-- ARROW_AWS_C_IO_BUILD_SHA256_CHECKSUM: 12b66510c3d9a4f7e9b714e9cfab2a5bf835f8b9ce2f909d20ae2a2128608c71
-- ARROW_AWS_C_MQTT_BUILD_VERSION: v0.8.4
-- ARROW_AWS_C_MQTT_BUILD_SHA256_CHECKSUM: 232eeac63e72883d460c686a09b98cdd811d24579affac47c5c3f696f956773f
-- ARROW_AWS_C_S3_BUILD_VERSION: v0.2.3
-- ARROW_AWS_C_S3_BUILD_SHA256_CHECKSUM: a00b3c9f319cd1c9aa2c3fa15098864df94b066dcba0deaccbb3caa952d902fe
-- ARROW_AWS_C_SDKUTILS_BUILD_VERSION: v0.1.6
-- ARROW_AWS_C_SDKUTILS_BUILD_SHA256_CHECKSUM: 8a2951344b2fb541eab1e9ca17c18a7fcbfd2aaff4cdd31d362d1fad96111b91
-- ARROW_AWS_CHECKSUMS_BUILD_VERSION: v0.1.13
-- ARROW_AWS_CHECKSUMS_BUILD_SHA256_CHECKSUM: 0f897686f1963253c5069a0e495b85c31635ba146cd3ac38cc2ea31eaf54694d
-- ARROW_AWS_CRT_CPP_BUILD_VERSION: v0.18.16
-- ARROW_AWS_CRT_CPP_BUILD_SHA256_CHECKSUM: 9e69bc1dc4b50871d1038aa9ff6ddeb4c9b28f7d6b5e5b1b69041ccf50a13483
-- ARROW_AWS_LC_BUILD_VERSION: v1.3.0
-- ARROW_AWS_LC_BUILD_SHA256_CHECKSUM: ae96a3567161552744fc0cae8b4d68ed88b1ec0f3d3c98700070115356da5a37
-- ARROW_AWSSDK_BUILD_VERSION: 1.10.55
-- ARROW_AWSSDK_BUILD_SHA256_CHECKSUM: 2d552fb1a84bef4a9b65e34aa7031851ed2aef5319e02cc6e4cb735c48aa30de
-- ARROW_AZURE_SDK_BUILD_VERSION: azure-core_1.10.3
-- ARROW_AZURE_SDK_BUILD_SHA256_CHECKSUM: dd624c2f86adf474d2d0a23066be6e27af9cbd7e3f8d9d8fd7bf981e884b7b48
-- ARROW_BOOST_BUILD_VERSION: 1.81.0
-- ARROW_BOOST_BUILD_SHA256_CHECKSUM: 9e0ffae35528c35f90468997bc8d99500bf179cbae355415a89a600c38e13574
-- ARROW_BROTLI_BUILD_VERSION: v1.0.9
-- ARROW_BROTLI_BUILD_SHA256_CHECKSUM: f9e8d81d0405ba66d181529af42a3354f838c939095ff99930da6aa9cdf6fe46
-- ARROW_BZIP2_BUILD_VERSION: 1.0.8
-- ARROW_BZIP2_BUILD_SHA256_CHECKSUM: ab5a03176ee106d3f0fa90e381da478ddae405918153cca248e682cd0c4a2269
-- ARROW_CARES_BUILD_VERSION: 1.17.2
-- ARROW_CARES_BUILD_SHA256_CHECKSUM: 4803c844ce20ce510ef0eb83f8ea41fa24ecaae9d280c468c582d2bb25b3913d
-- ARROW_CRC32C_BUILD_VERSION: 1.1.2
-- ARROW_CRC32C_BUILD_SHA256_CHECKSUM: ac07840513072b7fcebda6e821068aa04889018f24e10e46181068fb214d7e56
-- ARROW_GBENCHMARK_BUILD_VERSION: v1.7.1
-- ARROW_GBENCHMARK_BUILD_SHA256_CHECKSUM: 6430e4092653380d9dc4ccb45a1e2dc9259d581f4866dc0759713126056bc1d7
-- ARROW_GFLAGS_BUILD_VERSION: v2.2.2
-- ARROW_GFLAGS_BUILD_SHA256_CHECKSUM: 34af2f15cf7367513b352bdcd2493ab14ce43692d2dcd9dfc499492966c64dcf
-- ARROW_GLOG_BUILD_VERSION: v0.5.0
-- ARROW_GLOG_BUILD_SHA256_CHECKSUM: eede71f28371bf39aa69b45de23b329d37214016e2055269b3b5e7cfd40b59f5
-- ARROW_GOOGLE_CLOUD_CPP_BUILD_VERSION: v2.12.0
-- ARROW_GOOGLE_CLOUD_CPP_BUILD_SHA256_CHECKSUM: 8cda870803925c62de8716a765e03eb9d34249977e5cdb7d0d20367e997a55e2
-- ARROW_GRPC_BUILD_VERSION: v1.46.3
-- ARROW_GRPC_BUILD_SHA256_CHECKSUM: d6cbf22cb5007af71b61c6be316a79397469c58c82a942552a62e708bce60964
-- ARROW_GTEST_BUILD_VERSION: 1.11.0
-- ARROW_GTEST_BUILD_SHA256_CHECKSUM: b4870bf121ff7795ba20d20bcdd8627b8e088f2d1dab299a031c1034eddc93d5
-- ARROW_JEMALLOC_BUILD_VERSION: 5.3.0
-- ARROW_JEMALLOC_BUILD_SHA256_CHECKSUM: 2db82d1e7119df3e71b7640219b6dfe84789bc0537983c3b7ac4f7189aecfeaa
-- ARROW_LZ4_BUILD_VERSION: v1.9.4
-- ARROW_LZ4_BUILD_SHA256_CHECKSUM: 0b0e3aa07c8c063ddf40b082bdf7e37a1562bda40a0ff5272957f3e987e0e54b
-- ARROW_MIMALLOC_BUILD_VERSION: v2.0.6
-- ARROW_MIMALLOC_BUILD_SHA256_CHECKSUM: 9f05c94cc2b017ed13698834ac2a3567b6339a8bde27640df5a1581d49d05ce5
-- ARROW_NLOHMANN_JSON_BUILD_VERSION: v3.10.5
-- ARROW_NLOHMANN_JSON_BUILD_SHA256_CHECKSUM: 5daca6ca216495edf89d167f808d1d03c4a4d929cef7da5e10f135ae1540c7e4
-- ARROW_OPENTELEMETRY_BUILD_VERSION: v1.8.1
-- ARROW_OPENTELEMETRY_BUILD_SHA256_CHECKSUM: 3d640201594b07f08dade9cd1017bd0b59674daca26223b560b9bb6bf56264c2
-- ARROW_OPENTELEMETRY_PROTO_BUILD_VERSION: v0.17.0
-- ARROW_OPENTELEMETRY_PROTO_BUILD_SHA256_CHECKSUM: f269fbcb30e17b03caa1decd231ce826e59d7651c0f71c3b28eb5140b4bb5412
-- ARROW_ORC_BUILD_VERSION: 1.9.2
-- ARROW_ORC_BUILD_SHA256_CHECKSUM: 7f46f2c184ecefd6791f1a53fb062286818bd8710c3f08b94dd3cac365e240ee
-- ARROW_PROTOBUF_BUILD_VERSION: v21.3
-- ARROW_PROTOBUF_BUILD_SHA256_CHECKSUM: 2f723218f6cb709ae4cdc4fb5ed56a5951fc5d466f0128ce4c946b8c78c8c49f
-- ARROW_RAPIDJSON_BUILD_VERSION: 232389d4f1012dddec4ef84861face2d2ba85709
-- ARROW_RAPIDJSON_BUILD_SHA256_CHECKSUM: b9290a9a6d444c8e049bd589ab804e0ccf2b05dc5984a19ed5ae75d090064806
-- ARROW_RE2_BUILD_VERSION: 2022-06-01
-- ARROW_RE2_BUILD_SHA256_CHECKSUM: f89c61410a072e5cbcf8c27e3a778da7d6fd2f2b5b1445cd4f4508bee946ab0f
-- ARROW_SNAPPY_BUILD_VERSION: 1.1.10
-- ARROW_SNAPPY_BUILD_SHA256_CHECKSUM: 49d831bffcc5f3d01482340fe5af59852ca2fe76c3e05df0e67203ebbe0f1d90
-- ARROW_SUBSTRAIT_BUILD_VERSION: v0.27.0
-- ARROW_SUBSTRAIT_BUILD_SHA256_CHECKSUM: 4ed375f69d972a57fdc5ec406c17003a111831d8640d3f1733eccd4b3ff45628
-- ARROW_S2N_TLS_BUILD_VERSION: v1.3.35
-- ARROW_S2N_TLS_BUILD_SHA256_CHECKSUM: 9d32b26e6bfcc058d98248bf8fc231537e347395dd89cf62bb432b55c5da990d
-- ARROW_THRIFT_BUILD_VERSION: 0.16.0
-- ARROW_THRIFT_BUILD_SHA256_CHECKSUM: f460b5c1ca30d8918ff95ea3eb6291b3951cf518553566088f3f2be8981f6209
-- ARROW_UCX_BUILD_VERSION: 1.12.1
-- ARROW_UCX_BUILD_SHA256_CHECKSUM: 9bef31aed0e28bf1973d28d74d9ac4f8926c43ca3b7010bd22a084e164e31b71
-- ARROW_UTF8PROC_BUILD_VERSION: v2.7.0
-- ARROW_UTF8PROC_BUILD_SHA256_CHECKSUM: 4bb121e297293c0fd55f08f83afab6d35d48f0af4ecc07523ad8ec99aa2b12a1
-- ARROW_XSIMD_BUILD_VERSION: 9.0.1
-- ARROW_XSIMD_BUILD_SHA256_CHECKSUM: b1bb5f92167fd3a4f25749db0be7e61ed37e0a5d943490f3accdcd2cd2918cc0
-- ARROW_ZLIB_BUILD_VERSION: 1.2.13
-- ARROW_ZLIB_BUILD_SHA256_CHECKSUM: b3a24de97a8fdbc835b9833169501030b8977031bcb54b3b3ac13740f846ab30
-- ARROW_ZSTD_BUILD_VERSION: 1.5.5
-- ARROW_ZSTD_BUILD_SHA256_CHECKSUM: 9c4396cc829cfae319a6e2615202e82aad41372073482fce286fac78646d3ee4
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Looking for _M_ARM64
-- Looking for _M_ARM64 - not found
-- Looking for __SIZEOF_INT128__
-- Looking for __SIZEOF_INT128__ - found
-- Found Boost 1.82.0 at /opt/homebrew/lib/cmake/Boost-1.82.0
--   Requested configuration: REQUIRED
-- BoostConfig: find_package(boost_headers 1.82.0 EXACT CONFIG REQUIRED  HINTS /opt/homebrew/lib/cmake)
-- Found boost_headers 1.82.0 at /opt/homebrew/lib/cmake/boost_headers-1.82.0
-- Boost include dir: /opt/homebrew/include
-- Conan: Component target declared 'Snappy::snappy'
-- Found SnappyAlt: /Users/gangwu/.conan2/p/b/snappa4acd8e771160/p/lib/libsnappy.a
-- Providing CMake module for FindSnappyAlt as part of Arrow CMake package
-- pkg-config package for snappy that is used by arrow for static link isn't found
-- Conan: Component target declared 'brotli::brotlicommon'
-- Conan: Component target declared 'brotli::brotlidec'
-- Conan: Component target declared 'brotli::brotlienc'
-- Conan: Target declared 'brotli::brotli'
-- Providing CMake module for FindBrotliAlt as part of Arrow CMake package
-- Using pkg-config package for libbrotlidec that is used by arrow for static link
-- Using pkg-config package for libbrotlienc that is used by arrow for static link
-- Found OpenSSL: /opt/homebrew/opt/openssl@3/lib/libcrypto.dylib (found suitable version "3.1.4", minimum required is "1.0.2")
-- Providing CMake module for FindOpenSSLAlt as part of Arrow CMake package
-- Found OpenSSL Crypto Library: /opt/homebrew/opt/openssl@3/lib/libcrypto.dylib
-- Building with OpenSSL (Version: 3.1.4) support
-- Conan: Target declared 'glog::glog'
-- Conan: Target declared 'gflags::gflags'
-- Providing CMake module for FindGLOG as part of Arrow CMake package
-- pkg-config package for libglog that is used by arrow for static link isn't found
CMake Warning at cmake_modules/FindThriftAlt.cmake:56 (find_package):
  By not providing "FindThrift.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Thrift", but
  CMake did not find one.

  Could not find a package configuration file provided by "Thrift" (requested
  version 0.11.0) with any of the following names:

    ThriftConfig.cmake
    thrift-config.cmake

  Add the installation prefix of "Thrift" to CMAKE_PREFIX_PATH or set
  "Thrift_DIR" to a directory containing one of the above files.  If "Thrift"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  cmake_modules/ThirdpartyToolchain.cmake:307 (find_package)
  cmake_modules/ThirdpartyToolchain.cmake:1713 (resolve_dependency)
  CMakeLists.txt:542 (include)


-- Checking for module 'thrift'
--   Found thrift, version 0.19.0
-- Found ThriftAlt: /opt/homebrew/Cellar/thrift/0.19.0/lib/libthrift.dylib (found suitable version "0.19.0", minimum required is "0.11.0")
-- Providing CMake module for FindThriftAlt as part of Parquet CMake package
-- Using pkg-config package for thrift that is used by parquet for static link
-- Found ZLIB: /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib/libz.tbd (found version "1.2.11")
-- Providing CMake module for FindProtobufAlt as part of ArrowFlight CMake package
-- Using pkg-config package for protobuf that is used by arrow-flight for static link
-- Found protoc: /opt/homebrew/bin/protoc-3.21.12.0
-- Conan: Target declared 'jemalloc::jemalloc'
-- Providing CMake module for FindjemallocAlt as part of Arrow CMake package
-- Conan: Target declared 'xsimd'
-- xsimd found. Headers: /Users/gangwu/.conan2/p/xsimd3368e44453bd9/p/include
-- Using pkg-config package for zlib that is used by arrow for static link
-- Conan: Target declared 'LZ4::lz4_static'
-- Providing CMake module for Findlz4Alt as part of Arrow CMake package
-- Using pkg-config package for liblz4 that is used by arrow for static link
-- Conan: Component target declared 'zstd::libzstd_static'
-- Providing CMake module for FindzstdAlt as part of Arrow CMake package
-- Using pkg-config package for libzstd that is used by arrow for static link
-- Found Zstandard: zstd::libzstd_static
-- Providing CMake module for Findre2Alt as part of Arrow CMake package
-- Conan: Target declared 'BZip2::BZip2'
-- Conan: Including build module from '/Users/gangwu/.conan2/p/b/bzip2e465d44f60af5/p/lib/cmake/conan-official-bzip2-variables.cmake'
-- pkg-config package for bzip2 that is used by arrow for static link isn't found
CMake Warning at cmake_modules/FindgRPCAlt.cmake:25 (find_package):
  By not providing "FindgRPC.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "gRPC", but
  CMake did not find one.

  Could not find a package configuration file provided by "gRPC" (requested
  version 1.30.0) with any of the following names:

    gRPCConfig.cmake
    grpc-config.cmake

  Add the installation prefix of "gRPC" to CMAKE_PREFIX_PATH or set
  "gRPC_DIR" to a directory containing one of the above files.  If "gRPC"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  cmake_modules/ThirdpartyToolchain.cmake:307 (find_package)
  cmake_modules/ThirdpartyToolchain.cmake:4045 (resolve_dependency)
  CMakeLists.txt:542 (include)


-- Checking for module 'grpc++'
--   No package 'grpc++' found
CMake Error at /Users/gangwu/.conan2/p/cmakefc220a4e56e94/p/CMake.app/Contents/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find gRPCAlt (missing: GRPCPP_IMPORTED_LOCATION GRPC_CPP_PLUGIN)
  (Required is at least version "1.30.0")
Call Stack (most recent call first):
  /Users/gangwu/.conan2/p/cmakefc220a4e56e94/p/CMake.app/Contents/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake_modules/FindgRPCAlt.cmake:66 (find_package_handle_standard_args)
  cmake_modules/ThirdpartyToolchain.cmake:307 (find_package)
  cmake_modules/ThirdpartyToolchain.cmake:4045 (resolve_dependency)
  CMakeLists.txt:542 (include)


-- Configuring incomplete, errors occurred!

arrow/15.0.0: ERROR:
Package '92b89103d00005e6970e71f2ae5d7dfc138f73aa' build failed
arrow/15.0.0: WARN: Build folder /Users/gangwu/.conan2/p/b/arrow330afd2a1039a/b/build/Release
*********************************************************
Recipe 'arrow/15.0.0' cannot build its binary
It is possible that this recipe is not Conan 2.0 ready
If the recipe comes from ConanCenter, report it at https://github.com/conan-io/conan-center-index/issues
If it is your recipe, check if it is updated to 2.0
*********************************************************

ERROR: arrow/15.0.0: Error in build() method, line 534
	cmake.configure(build_script_folder=os.path.join(self.source_folder, "cpp"))
	ConanException: Error 1 while executing
```

</details>

**Comment by SpaceIm** [comment]
Indeed: https://github.com/conan-io/conan-center-index/pull/19380#discussion_r1382408724

**Comment by wgtmac** [comment]
> Indeed, it's a regression coming from PR which has migrated arrow to conan v2: [#19380 (comment)](https://github.com/conan-io/conan-center-index/pull/19380#discussion_r1382408724)
> 
> There is a fix in #21669 I think.

Yes, I have verified the fix works on my side.

**Comment by valgur** [comment]
Can be closed after #23163 was merged.
